### PR TITLE
Small Helm-related bug fixes.

### PR DESCRIPTION
**Pull request by hausdorff**

### CODE REVIEW

File: pack/nodejs/helm.ts [hunk 1]
```diff
@@ -78,6 +82,81 @@ export namespace v2 {
...
+            "Deployment",
+            "StatefulSet",
+            "Job",
+            "CronJob",
+            "Ingress",
+            "APIService"
+        ];
+
+        const ordering: { [key: string]: number } = {};
+        installOrder.forEach((kind, i) => {
```
Comment (pgavlin) [inline pack/nodejs/helm.ts]: Nit: `s/kind/_` or `s/installOrder[i]/kind` below.

File: pkg/gen/node-templates/provider.ts.mustache [hunk 2]
```diff
@@ -155,21 +163,38 @@ export namespace yaml {
...
-        if (!("kind" in obj && "apiVersion" in obj && "metadata" in obj)) {
-            return null;
+        if (!("kind" in obj && "apiVersion" in obj)) {
+            throw new Error(`Kubernetes resources require a kind and apiVersion: ${JSON.stringify(obj)}`)
         }
 
         const kind = obj["kind"];
         const apiVersion = obj["apiVersion"];
+
+        // Flatten `v1.List`. `v1.List` is an undocumented Kubernetes resource, and does not appear
```
Comment (swgillespie) [inline pkg/gen/node-templates/provider.ts.mustache]: wat

File: pack/nodejs/helm.ts [hunk 3]
```diff
@@ -61,14 +67,12 @@ export namespace v2 {
                 const chart = path.quotePath(nodepath.join(chartDir.name, config.chart));
                 const release = shell.quote([releaseName]);
                 const values = path.quotePath(overrides.name);
+                const defaultValues = path.quotePath(nodepath.join(chartDir.name, config.chart, "values.yaml"));
```
Comment (pgavlin) [inline pack/nodejs/helm.ts]: Is this file guaranteed to exist? What happens if it doesn't?

File: pack/nodejs/helm.ts [hunk 4]
```diff
@@ -78,6 +82,81 @@ export namespace v2 {
...
             }
         }
+
+        parseTemplate(
+            yamlStream: string,
+            config: ChartOpts
+        ): { [key: string]: pulumi.CustomResource } {
+            const objs = jsyaml
+                .safeLoadAll(yamlStream)
+                .filter(a => a != null && "kind" in a)
```
Comment (pgavlin) [inline pack/nodejs/helm.ts]: What does Helm do for resources that are missing `kind`?

File: pack/nodejs/helm.ts (hunk 3 above)
Comment (hausdorff) [reply pack/nodejs/helm.ts]: It's guaranteed to exist or it's not a valid chart.

File: pack/nodejs/helm.ts (hunk 4 above)
Comment (hausdorff) [reply pack/nodejs/helm.ts]: There is no such thing as a resource that does not have a `kind`.

Comment (hausdorff) [reply pack/nodejs/helm.ts]: This is basically to filter out empty YAML objects, which Helm can produce lots of.

**Comment by hausdorff** [comment]
@pgavlin has verbally signed off.

**Pull request merged**